### PR TITLE
PPF-67 - Improve frontend tooling

### PR DESCRIPTION
**Pull request by simon-debruijn**
### Changed

- Use typescript
- Add formatting
- Add linting
- Add tailwindcss
- Add ci jobs

---
Ticket: https://jira.uitdatabank.be/browse/PPF-67

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        "es2021": true
+    },
+    "extends": [
+        "eslint:recommended",
+        "plugin:react/recommended",
+        "plugin:@typescript-eslint/recommended",
+        "prettier"
+    ],
+    "overrides": [
+    ],
```
Comment (brampauwelyn) [inline .eslintrc.json:13]:
```suggestion
    "overrides": [],
```

File: resources/ts/Pages/Integrations/Create.tsx [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+          name="description"
+          id="description"
+          value={data.description}
+          onChange={(e) => setData('description', e.target.value)}
+        />
+        {errors.description && (
+          <div className="error">{errors.description}</div>
+        )}
+
+        <h4>Contact organisatie</h4>
```
Comment (brampauwelyn) [inline resources/ts/Pages/Integrations/Create.tsx:93]: Are we going to add something to handle translations?
I think it's useful to centralize all copy in 1 file.

File: resources/ts/Shared/Footer.tsx [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+
+export default function Footer() {
+  return (
+    <footer>
+      <p>Copyright 2022 publiq</p>
```
Comment (brampauwelyn) [inline resources/ts/Shared/Footer.tsx:6]: Make year dynamic ;) so you don't have to change it each year

File: resources/ts/Shared/Heading.tsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+};
+
+const levelToHeadingComponent = {
+  1: ({ children }: HeadingProps) => <h1 className="text-4xl">{children}</h1>,
+  2: ({ children }: HeadingProps) => <h2 className="text-2xl">{children}</h2>,
+  3: ({ children }: HeadingProps) => <h3 className="text-xl">{children}</h3>,
+  4: ({ children }: HeadingProps) => <h4 className="text-lg">{children}</h4>,
+  5: ({ children }: HeadingProps) => <h5 className="text-base">{children}</h5>,
+  6: ({ children }: HeadingProps) => <h6 className="text-sm">{children}</h6>,
+} as const;
```
Comment (brampauwelyn) [inline resources/ts/Shared/Heading.tsx:14]: Nice 👍

**Review by brampauwelyn — APPROVED** [review]

File: resources/ts/Shared/Footer.tsx (hunk 3 above)
Comment (simon-debruijn) [reply resources/ts/Shared/Footer.tsx:6]: You're totally right! This PR is focussing on the frontend tooling. In the actual frontend development we can change these things.

File: resources/ts/Pages/Integrations/Create.tsx (hunk 2 above)
Comment (simon-debruijn) [reply resources/ts/Pages/Integrations/Create.tsx:93]: Indeed I think we'll have to support multiple languages. Even if it's only dutch I'll add it to centralize the copy 👍🏻

File: resources/ts/Shared/Footer.tsx (hunk 3 above)
Comment (brampauwelyn) [reply resources/ts/Shared/Footer.tsx:6]: Ok 👍

**Review by LucWollants — APPROVED** [review]

**Pull request merged**